activerecord/test/cases/relation/or_test.rb

```diff
@@ -95,7 +95,7 @@ def test_or_with_incompatible_unscope
     end
 
     def test_or_when_grouping
-      groups = Post.where("id < 10").group("body").select("body, COUNT(*) AS c")
+      groups = Post.where("id < 10").group("body").select("body, COUNT(*) AS c").order("body DESC")
       expected = groups.having("COUNT(*) > 1 OR body like 'Such%'").to_a.map { |o| [o.body, o.c] }
       assert_equal expected, groups.having("COUNT(*) > 1").or(groups.having("body like 'Such%'")).to_a.map { |o| [o.body, o.c] }
```

Comment:
I know this result is potentially non-deterministic.
How about just use `.count` to test `.or` with different having clause?
Actually `.select("body, COUNT(*) AS c").to_a` is not desired usage for Active Record.

```suggestion
      groups = Post.where("id < 10").group("body")
      expected = groups.having("COUNT(*) > 1 OR body like 'Such%'").count
      assert_equal expected, groups.having("COUNT(*) > 1").or(groups.having("body like 'Such%'")).count
```

Reply:
I'm on board with this plan! Thanks for the suggestion :)